src/chart/generateCategoricalChart.tsx

```diff
@@ -240,7 +240,13 @@ const getTooltipContent = (
      */
     let data = child.props.data ?? chartData;
 
-    if (data && state.dataStartIndex + state.dataEndIndex !== 0) {
+    if (
+      data &&
+      state.dataStartIndex + state.dataEndIndex !== 0 &&
+      // https://github.com/recharts/recharts/issues/4717
+      // The data is sliced only when the active index is within the start/end index range.
+      state.dataEndIndex - state.dataStartIndex >= activeIndex
```

Comment:
This gets us back to the previous behavior as far as I know. 

I think it still won't play nice with brush but that's a different issue